Prior predictive guide

juanitorduz opened the pull request:
Many of our users (and most Bayesian practitioners, as it is a key component of the Bayesian workflow) have questions about prior predictive checks. We do this in many notebooks but go very fast. I'd like to suggest a very introductory guide to the topic. This could be a PyMC Example, but he tackles explicitly some prior specifications in PyMC-Marketing.

- [x] Introductory section
- [x] Simple example
- [x] MMM example 
- [x] References

<!-- readthedocs-preview pymc-marketing start -->
----
📚 Documentation preview 📚: https://pymc-marketing--1334.org.readthedocs.build/en/1334/

<!-- readthedocs-preview pymc-marketing end -->

wd60622 commented on docs/source/notebooks/general/prior_predictive.md:
What's the need for notebook and markdown file? How do they differ?

juanitorduz replied:
I'll remove the md file. Those are some notes and ideas [wip] 😉

juanitorduz commented:
@wd60622 I am using the plotting functions from the mmm components! They are amazing! We should promote them more!

wd60622 commented:
Yeah, I like them too. Convenient workflow for prior and posterior. I think they should be on all component based items with same / similar API.

juanitorduz commented:
This one is ready for a first review round :)

juanitorduz commented:
<p>This is a great point. I think I will add it as a remark if that is ok :)</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1334/discussion/#comment-2577470414'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

juanitorduz commented:
<p>good catch! </p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1334/discussion/#comment-2577470471'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

juanitorduz commented:
<p>done!</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1334/discussion/#comment-2577470446'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

juanitorduz commented:
<p>Great point! Added a fererence~</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1334/discussion/#comment-2577470487'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

juanitorduz commented:
<p>I added a Truncated Normal :)</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1334/discussion/#comment-2577470546'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

juanitorduz commented:
<p>done!</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1334/discussion/#comment-2577820974'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

juanitorduz commented:
<p>cool!</p>

---

 <a href='https://app.reviewnb.com/pymc-labs/pymc-marketing/pull/1334/discussion/#comment-2577821002'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

juanitorduz commented:
Thank you @wd60622 and @ErikRingen great feedback! I think I have addressed all your comments (If not please let me know). This one is ready for a second review round 🙏

ErikRingen commented:
> Thank you @wd60622 and @ErikRingen great feedback! I think I have addressed all your comments (If not please let me know). This one is ready for a second review round 🙏

Looks really good to me! It is such a big topic of course we could think of more to add but this seems like the right length/amount of detail for a first tutorial on prior predictions. Something perhaps for a follow-up notebook is the different types of informative/regularizing priors, the most important distiction I think being informative with respective to magnitude (i.e., a priori we expect the effect of be small) vs informative with respective to sign (i.e., a priori we expect the effect to be positive). When teaching I often reference the "5 levels" of priors from the stan devs: https://github.com/stan-dev/stan/wiki/prior-choice-recommendations

juanitorduz commented:
> > Thank you @wd60622 and @ErikRingen great feedback! I think I have addressed all your comments (If not please let me know). This one is ready for a second review round 🙏
> 
> Looks really good to me! It is such a big topic of course we could think of more to add but this seems like the right length/amount of detail for a first tutorial on prior predictions. Something perhaps for a follow-up notebook is the different types of informative/regularizing priors, the most important distiction I think being informative with respective to magnitude (i.e., a priori we expect the effect of be small) vs informative with respective to sign (i.e., a priori we expect the effect to be positive). When teaching I often reference the "5 levels" of priors from the stan devs: https://github.com/stan-dev/stan/wiki/prior-choice-recommendations

Thanks for the comments! Indeed, this is a very interesting topic! I will add the Stan link as a reference (look great!). If you have up for it, you would write a follow-up notebook or extend this one (on iteration) based on your valuable experience teaching these topics 💪 !

juanitorduz commented:
@ErikRingen would you mind approving this PR then ? 🤗

juanitorduz commented:
yay! thanks you for the reviews!
